pkg/support/runsupportbundle.go

```diff
@@ -1,172 +1,135 @@
+		WithExternalEtcd(opts.ClusterSpec.Spec.ExternalEtcdConfiguration).
+		WithDatacenterConfig(opts.ClusterSpec.Spec.DatacenterRef).
+		WithDefaultAnalyzers().
+		WithDefaultCollectors()
+
+	err := b.WriteBundleConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error writing bundle config: %v", err)
 	}
-	return b.WithDefaultAnalyzers().WithDefaultCollectors()
+
+	return b, nil
 }
 
-func NewBundleConfig(spec *cluster.Spec, af AnalyzerFactory, cf CollectorFactory) *EksaDiagnosticBundle {
+func NewDiagnosticBundleDefault(af AnalyzerFactory, cf CollectorFactory) *EksaDiagnosticBundle {
 	b := &EksaDiagnosticBundle{
-		bundle: &v1beta2.SupportBundle{
+		bundle: &supportBundle{
 			TypeMeta: metav1.TypeMeta{
 				Kind:       "SupportBundle",
-				APIVersion: "troubleshoot.sh/v1beta2",
+				APIVersion: troulbeshootApiVersion,
 			},
 			ObjectMeta: metav1.ObjectMeta{
-				Name: fmt.Sprintf("%sBundle", spec.Name),
+				Name: "defaultBundle",
 			},
-			Spec: v1beta2.SupportBundleSpec{},
+			Spec: supportBundleSpec{},
 		},
 		analyzerFactory:  af,
 		collectorFactory: cf,
 	}
-	return b.
-		WithGitOpsConfig(spec.GitOpsConfig).
-		WithOidcConfig(spec.OIDCConfig).
-		WithExternalEtcd(spec.Spec.ExternalEtcdConfiguration).
-		WithDatacenterConfig(spec.Spec.DatacenterRef).
-		WithDefaultAnalyzers().
-		WithDefaultCollectors()
+	return b.WithDefaultAnalyzers().WithDefaultCollectors()
 }
 
-func (e *EksaDiagnosticBundle) CollectBundleFromSpec(sinceTimeValue *time.Time) (string, error) {
-	k8sConfig, err := k8sutil.GetRESTConfig()
-	if err != nil {
-		return "", fmt.Errorf("failed to convert kube flags to rest config: %v", err)
+func NewDiagnosticBundleCustom(opts EksaDiagnosticBundleOpts) *EksaDiagnosticBundle {
+	return &EksaDiagnosticBundle{
+		bundlePath:       opts.BundlePath,
+		analyzerFactory:  opts.AnalyzerFactory,
+		collectorFactory: opts.CollectorFactory,
+		client:           opts.Client,
+		kubeconfig:       opts.Kubeconfig,
 	}
+}
 
-	progressChan := make(chan interface{})
-	go func() {
-		var lastMsg string
-		for {
-			msg := <-progressChan
-			switch msg := msg.(type) {
-			case error:
-				logger.Info(fmt.Sprintf("\r * %v", msg))
-			case string:
-				if lastMsg != msg {
-					logger.Info(fmt.Sprintf("\r \033[36mCollecting support bundle\033[m %v", msg))
-					lastMsg = msg
-				}
-			}
-		}
-	}()
-
-	collectorCB := func(c chan interface{}, msg string) {
-		c <- msg
-	}
-	additionalRedactors := &v1beta2.Redactor{}
-	createOpts := supportbundle.SupportBundleCreateOpts{
-		CollectorProgressCallback: collectorCB,
-		KubernetesRestConfig:      k8sConfig,
-		ProgressChan:              progressChan,
-		SinceTime:                 sinceTimeValue,
+func (e *EksaDiagnosticBundle) CollectAndAnalyze(ctx context.Context, sinceTimeValue *time.Time) error {
+	archivePath, err := e.client.Collect(ctx, e.bundlePath, sinceTimeValue, e.kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to Collect and Analyze support bundle: %v", err)
```

Comment:
do you think the error msg here can just say "failed to collect support bundle", and then the one after the call to `e.client.Analyze(ctx, e.bundlePath, archivePath)` can say "failed to analyze support bundle"?